### PR TITLE
The big and scary fingerprint update.

### PR DESCRIPTION
**Pull request by beorn7**
See commit descriptions.

@juliusv

### CODE REVIEW

File: storage/local/persistence.go [hunk 1]
```diff
@@ -36,6 +38,11 @@ import (
 )
 
 const (
+	// Version of the storage, as it can be found in the version file.
+	// Increment to protect against icompatible changes.
```
Comment (juliusv) [inline storage/local/persistence.go]: "icompatible" -> "incompatible"

File: storage/local/persistence.go [hunk 2]
```diff
@@ -36,6 +38,11 @@ import (
 )
 
 const (
+	// Version of the storage, as it can be found in the version file.
```
Comment (juliusv) [inline storage/local/persistence.go]: I think that comma is not necessary...

File: storage/local/persistence.go [hunk 3]
```diff
@@ -114,10 +121,49 @@ type persistence struct {
...
 // newPersistence returns a newly allocated persistence backed by local disk storage, ready to use.
 func newPersistence(basePath string, chunkLen int, dirty bool) (*persistence, error) {
-	if err := os.MkdirAll(basePath, 0700); err != nil {
+	dirtyPath := filepath.Join(basePath, dirtyFileName)
+	versionPath := filepath.Join(basePath, versionFileName)
+
+	if file, err := os.Open(versionPath); err == nil {
+		defer file.Close()
+		data := make([]byte, 8)
+		n, err := file.Read(data)
```
Comment (juliusv) [inline storage/local/persistence.go]: If you use http://golang.org/pkg/io/ioutil/#ReadFile you'll get the whole file returned guaranteed.

**Comment by juliusv** [comment]
:+1: modulo comments

File: storage/local/persistence.go (hunk 1 above)
Comment (beorn7) [inline storage/local/persistence.go]: DOne.

File: storage/local/persistence.go (hunk 2 above)
Comment (beorn7) [inline storage/local/persistence.go]: Done.

File: storage/local/persistence.go (hunk 3 above)
Comment (beorn7) [inline storage/local/persistence.go]: Ah, right. I was looking for that and couldn't find it. Same for ReadDir. Will incorporate those.

**Pull request merged**